frontend/public/components/utils/status-icon.tsx

```diff
@@ -54,37 +74,40 @@ export const StatusIcon: React.FunctionComponent<StatusIconProps> = ({status, sp
     case 'Enabled':
     case 'Ready':
     case 'Up to date':
-      return <Icon type="pf" name="ok" className={className} />;
+      return <GreenCheckCircleIcon className={className} />;
 
     case 'Info':
-      return <Icon type="pf" name="info" className={className} />;
+      return <InfoCircleIcon className={className} />;
 
     case 'Unknown':
-      return <Icon type="pf" name="unknown" className={className} />;
+      return <UnknownIcon className={className} />;
 
     default:
       return null;
   }
 };
 
-
-export const StatusIconAndText: React.FunctionComponent<StatusIconAndTextProps> = ({status, spin, iconName}) => {
+export const StatusIconAndText: React.FunctionComponent<StatusIconAndTextProps> = ({status, spin, icon}) => {
   if (!status){
     return <span>-</span>;
   }
 
-  return <span className="co-icon-and-text"><StatusIcon status={status} spin={spin} additionalIconClassName="co-icon-and-text__icon" iconName={iconName} /><CamelCaseWrap value={status} /></span>;
+  return <span className="co-icon-and-text"><StatusIcon status={status} spin={spin} additionalIconClassName="co-icon-and-text__icon" icon={icon} /><CamelCaseWrap value={status} /></span>;
+};
+
+export type ColoredIconProps = {
+  className?: string;
 };
 
 export type StatusIconProps = {
   status?: string;
   additionalIconClassName?: any;
   spin?: boolean;
-  iconName?: string;
+  icon?: React.ReactElement;
```

Comment:
Generally you want `React.ReactNode`

```suggestion
  icon?: React.ReactNode;
```

Reply:
This change results in the following error:
```ERROR in /Users/rhamilto/Git/console/frontend/public/components/utils/status-icon.tsx(30,31):
TS2345: Argument of type 'string | number | true | {} | ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)>) | (new (props: any) => Component<any, any, any>)> | ReactNodeArray | ReactPortal' is not assignable to parameter of type 'ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)>) | (new (props: any) => Component<any, any, any>)>'.
  Type 'string' is not assignable to type 'ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)>) | (new (props: any) => Component<any, any, any>)>'```

Reply:
OK, disregard my suggestion :)